### ISSUE TITLE
No logs flush on SIGTERM caused by back-off mechanism

### ISSUE THREAD

**Issue by radekkujawa** [fluent/fluent-bit#5960]
## Bug Report

**Describe the bug**
When Fluentbit output loses connectivity to target, exponential back-off kicks in. Once it is active, buffered logs are not being sent on SIGTERM signal.

**To Reproduce**
Tested with Fluentbit and Loki as a target.
- during normal, working operation simulate connectivity loss to output target
- keep generating logs, so they are stored in the FB buffer
- wait for `[engine] failed to flush chunk '1-31r52845t294189.flb', retry in XXX seconds` error message and immediately send SIGTERM signal
- even though `Grace` and `Flush` config are in place, logs are not being flushed to Loki destination

Logs
```
2022-08-23T13:07:42.370+02:00 | [2022/08/23 13:07:42] [error] [upstream] connection #65 to CENSORED:3100 timed out after 10 seconds
2022-08-23T13:07:42.380+02:00 | [2022/08/23 13:07:42] [error] [output:loki:loki.0] no upstream connections available
2022-08-23T13:07:42.390+02:00 | [2022/08/23 13:07:42] [ warn] [engine] failed to flush chunk '1-31r52845t294189.flb', retry in 195 seconds: task_id=0, input=forward.1 > output=loki.0 (out_id=0)
2022-08-23T13:09:45.370+02:00 | [2022/08/23 11:09:45] [engine] caught signal (SIGTERM)
2022-08-23T13:09:45.370+02:00 | [2022/08/23 11:09:45] [ info] [input] pausing forward.1
2022-08-23T13:09:45.370+02:00 | [2022/08/23 11:09:45] [ info] [input] pausing forward.2
2022-08-23T13:09:45.370+02:00 | [2022/08/23 11:09:45] [ info] [input] pausing storage_backlog.3
2022-08-23T13:09:45.370+02:00 | [2022/08/23 11:09:45] [ warn] [engine] service will shutdown in max 30 seconds
2022-08-23T13:09:46.295+02:00 | [2022/08/23 11:09:46] [ info] [engine] service has stopped (0 pending tasks)
2022-08-23T13:09:46.295+02:00 | [2022/08/23 11:09:46] [ info] [output:null:null.1] thread worker #0 stopping...
2022-08-23T13:09:46.295+02:00 | [2022/08/23 11:09:46] [ info] [output:null:null.1] thread worker #0 stopped
2022-08-23T13:09:46.295+02:00 | [2022/08/23 11:09:46] [ info] [output:cloudwatch_logs:cloudwatch_logs.2] thread worker #0 stopping...
2022-08-23T13:09:46.295+02:00 | [2022/08/23 11:09:46] [ info] [output:cloudwatch_logs:cloudwatch_logs.2] thread worker #0 stopped
```
**Expected behavior**
After SIGTERM signal is received, forcefully try to send logs from buffer to target, bypassing back-off mechanism.

**Your Environment**
* Version used:
```
bash-4.2# ./fluent-bit --version
Fluent Bit v1.9.7
Git commit: 265783ebe99590cf1687d2f48b300f58febb883d
```

* Configuration:
```
[SERVICE]
    Log_Level info
    Parsers_File /fluent-bit/etc/parsers.conf
    Flush 1
    Grace 30
    storage.path              /tmp/flb-storage/
    storage.sync              normal
    storage.checksum          off
    storage.backlog.mem_limit 15M

[FILTER]
    name                  multiline
    match                 *
    multiline.key_content log
    mode                  partial_message

[FILTER]
    Name parser
    Match **
    Parser json
    Key_Name log
    Reserve_Data True

[OUTPUT]
    Name loki
    Match *
    Host ${LOKI_ADDRESS}
    Labels job=${NAME}, env=${AWS_ENV}
    line_format json
    remove_keys container_name,ecs_task_arn
    label_keys $ecs_task_definition,$source,$ecs_cluster,$container_id
    Retry_Limit no_limits
    storage.total_limit_size  1024M
```

* Environment name and version: AWS ECS
* Server type and version: AWS ECS Fargate
* Filters and plugins: Native Loki output, CloudWatch output

Cheers!

**Comment by leonardo-albertovich** [comment]
I think #5105 addresses this issue.

**Comment by radekkujawa** [comment]
Thanks @leonardo-albertovich it looks like this one `All retries are rescheduled immediately when the engine first processes a shutdown request` will save the day. Do you know in which version it's going to be released?

Thanks!

**Comment by leonardo-albertovich** [comment]
Not really, I think it's possible that it's included in 2.0 but I could be wrong.
